maven-plugin/src/main/java/org/openmrs/maven/plugins/Setup.java

```diff
@@ -155,183 +157,161 @@ public Setup(AbstractTask other){
+        // *after* having initialized server.isH2Supported in `setupDatabase` above.
+        if (distroProperties == null) {
+            distroProperties = createDistroForPlatform(server);
+        }
+        distroProperties.saveTo(server.getServerDirectory());
 
-            String platformVersion = server.getPlatformVersion();
-            Version version = new Version(platformVersion);
-            if (platformVersion.startsWith("1.")) {
-                wizard.showMessage("Note: JDK 1.7 is needed for platform version " + platformVersion + ".");
-            }
-            else if (version.getMajorVersion() == 2 && version.getMinorVersion() < 4) {
-            	wizard.showMessage("Note: JDK 1.8 is needed for platform version " + platformVersion + ".");
-            }
-            else {
-                wizard.showMessage("Note: JDK 1.8 or above is needed for platform version " + platformVersion + ".");
-            }
+        setJdk(server);
 
-            wizard.promptForJavaHomeIfMissing(server);
-            server.setValuesFromDistroPropertiesModules(
-                    distroProperties.getWarArtifacts(distroHelper, server.getServerDirectory()),
-                    distroProperties.getModuleArtifacts(distroHelper, server.getServerDirectory()),
-                    distroProperties
-            );
-            server.setUnspecifiedToDefault();
-            server.save();
+        server.setValuesFromDistroPropertiesModules(
+                distroProperties.getWarArtifacts(distroHelper, server.getServerDirectory()),
+                distroProperties.getModuleArtifacts(distroHelper, server.getServerDirectory()),
+                distroProperties
+        );
+        server.setUnspecifiedToDefault();
+        server.save();
+    }
 
-            return serverPath.getPath();
-        } catch (Exception e) {
-            FileUtils.deleteQuietly(serverPath);
-            throw new MojoExecutionException("Failed to setup server", e);
+    private void setJdk(Server server) {
+        String platformVersion = server.getPlatformVersion();
+        Version version = new Version(platformVersion);
+        if (platformVersion.startsWith("1.")) {
+            wizard.showMessage("Note: JDK 1.7 is needed for platform version " + platformVersion + ".");
+        }
+        else if (version.getMajorVersion() == 2 && version.getMinorVersion() < 4) {
+            wizard.showMessage("Note: JDK 1.8 is needed for platform version " + platformVersion + ".");
         }
+        else {
+            wizard.showMessage("Note: JDK 1.8 or above is needed for platform version " + platformVersion + ".");
+        }
+
+        wizard.promptForJavaHomeIfMissing(server);
     }
     
     private void installOWAs(Server server, DistroProperties distroProperties) throws MojoExecutionException {
     	if (distroProperties != null) {
     		File owasDir = new File(server.getServerDirectory(), "owa");
-    		owasDir.mkdirs();
+    		owasDir.mkdir();
```

Comment:
Previously, `installOWAs` was being called all over the place, and was accidentally responsible for creating the server directory. Using `mkdir` would have made it obvious that this was happening.